sql/src/main/java/org/apache/druid/sql/calcite/aggregation/builtin/EarliestLatestAnySqlAggregator.java

```diff
@@ -340,5 +345,47 @@ private static class EarliestLatestSqlAggFunction extends SqlAggFunction
           Optionality.FORBIDDEN
       );
     }
+
+    @Override
+    public SqlNode rewriteCall(
+        SqlValidator validator,
+        SqlCall call
+    )
+    {
+      // Rewrite EARLIEST to EARLIEST_BY and LATEST to LATEST_BY to make
+      // reference to __time column explicit so that Calcite tracks it
+
+      List<SqlNode> operands = call.getOperandList();
+
+      SqlParserPos pos = call.getParserPosition();
+
+      SqlAggFunction aggFunction;
+
+      switch (getName()) {
+        case "EARLIEST":
+          aggFunction = EarliestLatestBySqlAggregator.EARLIEST_BY.calciteFunction();
+          break;
+        case "LATEST":
+          aggFunction = EarliestLatestBySqlAggregator.LATEST_BY.calciteFunction();
+          break;
+        default:
+          return call;
+      }
+
+      switch (operands.size()) {
+        case 1:
+          return aggFunction.createCall(pos, operands.get(0), new SqlIdentifier(
+              "__time", pos));
+        case 2:
+          return aggFunction.createCall(pos, operands.get(0), new SqlIdentifier(
```

Comment:
I wonder if we should validate the new call created as well - since we're hard coding the `__time` now in the function call. It may or may not be present in the aggregation scope - which is actually a bad thing that the LATEST/EARLIEST invocations have.

Reply:
The new call will be validated post the rewrite. But there was still the issue of validator treating rewritten query as originating from the user and flagging unrelatable `col __time not found in any table (row x, col y)` error. I've now addressed it by introducing a rewrite tracking boolean in the validator.